src/java.base/share/classes/java/lang/constant/ClassDesc.java

```diff
@@ -312,9 +336,9 @@ default ClassDesc componentType() {
     default String packageName() {
         if (!isClassOrInterface())
             return "";
-        String className = internalToBinary(ConstantUtils.dropFirstAndLastChar(descriptorString()));
-        int index = className.lastIndexOf('.');
-        return (index == -1) ? "" : className.substring(0, index);
+        String desc = descriptorString();
+        int index = desc.lastIndexOf('/');
+        return (index == -1) ? "" : internalToBinary(desc.substring(1, index - 1));
```

Comment:
```suggestion
        return (index == -1) ? "" : internalToBinary(desc.substring(1, index));
```

Reply:
Thanks! Verified there were tests covering this.